quartz_solar_forecast/data.py

```diff
@@ -206,9 +207,26 @@ def make_pv_data(site: PVSite, ts: pd.Timestamp) -> xr.Dataset:
             live_generation_kw = get_givenergy_data()
         except Exception as e:
             print(f"Error retrieving GivEnergy data: {str(e)}")
+    elif site.inverter_type == 'solarman':
+        try:
+            end_date = datetime.now()
+            start_date = end_date - timedelta(weeks=1)
+            solarman_data = get_solarman_data(start_date, end_date)
+            print("DATA: ", solarman_data)
+            # Filter data for the specified timestamp or find the nearest past timestamp
+            filtered_data = solarman_data[solarman_data['timestamp'] <= ts].sort_values('timestamp', ascending=False)
+            
+            if not filtered_data.empty:
+                live_generation_kw = filtered_data.head(1)  # Get the most recent available data point as DataFrame
```

Comment:
Why do we use just the last data point? What granularity of data do we get here and are we throwing away useful historical power data by doing this?

Reply:
> Why do we use just the last data point? What granularity of data do we get here and are we throwing away useful historical power data by doing this?

We get the data from start time to end time with a 5 min interval between each , this filtered data is just added as a precautionary measure so that we are 100% sure that we get data before the current time only

Reply:
Sorry I am talking about line 220?

Reply:
> Sorry I am talking about line 220?

Oh that, thats a fix i made because it doesnt get updated at the current time most often, and the result is an array with the first element(arr[0]) being null more times than not, so to be safe im taking the data 5 mins before the latest data

Reply:
The way I am reading this is that we initially request a weeks worth of data, which we get back in 5 minute increments, and then we only take 1 data point from all of that? I may have misread or misunderstood so do correct me if I have, but if not then this seems wasteful? That extra data could be useful for the model as it doesn't just use the last data point

Reply:
> The way I am reading this is that we initially request a weeks worth of data, which we get back in 5 minute increments, and then we only take 1 data point from all of that? I may have misread or misunderstood so do correct me if I have, but if not then this seems wasteful? That extra data could be useful for the model as it doesn't just use the last data point

oh no no, we just start from the 2nd element of the arr, ie [1]

Reply:
`.head(n) `will just return the first n rows no? https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.head.html and the `filtered_data` is a dataframe with potentially many rows?

Reply:
> `.head(n) `will just return the first n rows no? https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.head.html and the `filtered_data` is a dataframe with potentially many rows?

Thanks for pointing this out, realised i hadnt pushed the change for this-thought i could reduce the api calls similar to solis, but turns out it's real quick